pkg/framework/context/context.go

```diff
@@ -20,7 +19,7 @@ type Provider struct {
 }
 
 // New instantiated new context provider
-func New(opts ...ProviderOption) (*Provider, error) {
+func New(newProtocolSvc []api.NewProtocolSvc, opts ...ProviderOption) (*Provider, error) {
```

Comment:
`context.Provider` already has apiHandlers. Could these `api.NewProtocolSvc` be made into `ProviderOption` instead?

Reply:
done